pkg/ottl/parser.go

```diff
@@ -141,7 +154,50 @@ func (p *Parser[K]) ParseStatement(statement string) (*Statement[K], error) {
 	}, nil
 }
 
+// ParseConditions parses string conditions into a ottl.Condition slice ready for execution.
+// Returns a slice of conditions and a nil error on successful parsing.
+// If parsing fails, returns an empty slice  with a multierr error containing
+// an error per failed condition.
+func (p *Parser[K]) ParseConditions(conditions []string) ([]*Condition[K], error) {
+	parsedConditions := make([]*Condition[K], 0, len(conditions))
+	var parseErr error
+
+	for _, condition := range conditions {
+		ps, err := p.ParseCondition(condition)
+		if err != nil {
+			parseErr = multierr.Append(parseErr, fmt.Errorf("unable to parse OTTL condition %q: %w", condition, err))
```

Comment:
I think we're trying to move away from multierr - maybe ok to add since it's used already

Reply:
Updated